node/src/protocol/triple.rs

```diff
@@ -100,6 +106,15 @@ impl TripleManager {
         Ok(())
     }
 
+    pub fn generate_pile_by_bandwidth(&mut self, nodes: usize) -> Result<(), InitializationError> {
+        let pile = calc_optimal_pile(DEFAULT_MAX_MESSAGES, nodes);
+        tracing::info!(nodes, pile, "generating pile of triples");
+        for _ in 0..pile.min(DEFAULT_MAX_PILE) {
```

Comment:
Is this going to lock up the node or is this running on it's own thread?

Reply:
@itegulov correct me if I'm wrong, but it shouldn't lock up the node, as the triple protocol will just yield back a bunch of messages to be sent once poked, so the crypto/consensus loop should keep on continuing. As far as I saw, probably the only case that might've been bad is if it wants to generate a certain triple with a TripleId but the stockpile of triple has priority and the node needs to wait longer to generate the signature in that case